library/alloc/src/collections/btree/map/entry.rs

```diff
@@ -49,21 +49,21 @@ pub struct VacantEntry<
     'a,
     K,
     V,
-    #[unstable(feature = "allocator_api", issue = "32838")] A: Allocator = Global,
+    #[unstable(feature = "allocator_api", issue = "32838")] A: Allocator + Clone = Global,
 > {
     pub(super) key: K,
     /// `None` for a (empty) map without root
     pub(super) handle: Option<Handle<NodeRef<marker::Mut<'a>, K, V, marker::Leaf>, marker::Edge>>,
     pub(super) dormant_map: DormantMutRef<'a, BTreeMap<K, V, A>>,
 
-    pub(super) alloc: &'a A,
+    pub(super) alloc: A,
```

Comment:
Should this be `ManuallyDrop<A>`? It is in the btree itself, presumably due to drop ordering concerns. (If `A: Copy` then this would also not be a problem...)